### PR TITLE
Accelerate PADDLE_ENFORCE

### PR DESCRIPTION
**Pull request by velconia**

### CODE REVIEW

**Review by phlrain — DISMISSED** [review]
LGTM

**Review by phlrain — DISMISSED** [review]

File: paddle/fluid/framework/operator.h [hunk 1]
```diff
@@ -60,7 +62,11 @@ constexpr char kNewGradSuffix[] = "@NEWGRAD@";
 extern std::vector<std::tuple<platform::Place, LibraryType>> kKernelPriority;
 
 inline std::string GradVarName(const std::string& var_name) {
-  return var_name + kGradVarSuffix;
+  std::string result;
+  result.reserve(var_name.size() + kGradVarSuffixSize);
+  result += var_name;
```
Comment (typhoonzero) [inline paddle/fluid/framework/operator.h:67]: does `+=` still make a temp object here, can use `append` instead?

Comment (velconia) [reply paddle/fluid/framework/operator.h:67]: ```+=``` is actually same with ```append```

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**